src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/visits/VisitsMigrationService.kt

```diff
@@ -76,36 +76,44 @@ class VisitsMigrationService(
   fun migrateVisit(context: MigrationContext<VisitId>) =
     visitMappingService.findNomisVisitMapping(context.body.visitId)
       ?.run {
-        log.error("Will not migrate visit since it is migrated already, NOMIS id is ${context.body.visitId}, VSIP id is ${this.vsipId} as part migration ${this.label ?: "NONE"} (${this.mappingType})")
+        log.info("Will not migrate visit since it is migrated already, NOMIS id is ${context.body.visitId}, VSIP id is ${this.vsipId} as part migration ${this.label ?: "NONE"} (${this.mappingType})")
       }
       ?: run {
         val nomisVisit = nomisApiService.getVisit(context.body.visitId)
-        log.info("Migrating visit {}", nomisVisit)
-        val room = nomisVisit.agencyInternalLocation?.let {
+        nomisVisit.agencyInternalLocation?.let {
           visitMappingService.findRoomMapping(
             agencyInternalLocationCode = nomisVisit.agencyInternalLocation.description,
             prisonId = nomisVisit.prisonId
           )
+        }?.run {
+          val vsipVisit = visitsService.createVisit(
+            mapNomisVisit(nomisVisit, this)
+          )
+          visitMappingService.createNomisVisitMapping(
+            nomisVisitId = nomisVisit.visitId,
+            vsipVisitId = vsipVisit.visitId,
+            migrationId = context.migrationId
+          )
+        } ?: run {
+          log.error("Could not find room mapping for visit ${nomisVisit.visitId} with prison ${nomisVisit.prisonId} and agency internal location ${nomisVisit.agencyInternalLocation?.description}")
```

Comment:
confirmed with VSIP Team, no mapping should result in error

Reply:
I thought it would just log the error and not throw an exception - we don't want it on the DLQ do we?

Reply:
I was thinking of sticking on DLQ so we have a record, and VSIP can then decide to either fix forward or discard all DLQ messages